tools/clang/test/CodeGenSPIRV/spirv.debug.opline.hlsl

```diff
@@ -0,0 +1,33 @@
+// Run: %dxc -T ps_6_0 -E main -Zi
+
+// CHECK:      [[file:%\d+]] = OpString
+// CHECK-SAME: spirv.debug.opline.hlsl
+
+Texture2D    MyTexture;
+SamplerState MySampler;
+
+uint foo(uint val) {
+  return val;
+}
+
+// Note that we do two passes for debuging info: first preprocessing and then
+// compiling the preprocessed source code.
+// Because the preprocessor prepends a "#line 1 ..." line to the whole file,
+// the compliation sees line numbers incremented by 1.
+
+float4 main(uint val : A) : SV_Target {
+  // CHECK:      OpLine [[file]] 23 12
```

Comment:
Shouldn't this be line 22?

Reply:
It's tricky. See the comment in the above.

Reply:
We are not using the exact original file in the compilation, but a preprocessed version of it.

Reply:
Ah Sorry I don't know why i didn't see the comment before. Thanks

Reply:
You can try to do `-P` over a normal HLSL source. If it does not include other files, there is one `#line 1 ...` prepended to it.